### ISSUE TITLE
Registering and deleting terms with the same name

### ISSUE THREAD

**Issue by imutaka** [qtranslate/qtranslate-xt#1070]
Hello.

I am having a problem with the translation data of terms.
When I have the same term name in different levels, if I delete the term in the lower level, the translation data of the term in the upper level will be lost.

Also, it is not possible to register the same name in the main language, but have different names in another languages.

The data seems to be stored in wp_options as translate_term_name, which is serialized, but I think this is happening because the key is the term name of the default language.
Is it possible to use term_id as a key?

Thanks.

**Comment by imutaka** [comment]
Hi.

slug is different.

The following data is available.

wp_terms
|term_id|name|slag|
|---|---|---|
|1|Colognes|colognes|
|2|Candles|candles|
|3|Citrus|co_citrus|
|4|Citrus|ca_citrus|

wp_options
|option_id|option_name|option_value|
|---|---|---|
|99|qtranslate_term_name|a:3:{s:8: "Colognes";a:2:{s:2: "en";s:8: "Colognes";s:2: "ja";s:9: "コロン";}s:7: "Candles";a:2:{s:2: "en";s:7:" Candles";s:2: "ja";s:15: "キャンドル";}s:6: "Citrus";a:2:{s:2: "en";s:6: "Citrus";s:2: "ja";s:12: "シトラス";}}|
```
$terms = [
    "Colognes" => [
        "en" => "Colognes",
        "ja" => "コロン"
    ],
    "Candles" => [
        "en" => "Candles",
        "ja" => "キャンドル"
    ],
    "Citrus" => [
        "en" => "Citrus",
        "ja" => "シトラス"
    ]
];
```

The problem is that when term_id=3 is deleted, the translation information for the term with the key "Citrus" is deleted, and so is the translation information for term_id=4.

I think this is caused by a bad way of saving term information in qtranslate, not in WordPress.

Thanks.

**Comment by MK-RD** [comment]
Yep, your analysis is **absolutely correct** and I can't believe I hadn't noticed that before.

With the **initial change** to "**term_id**" for keys, existing keys (name) should also be changed accordingly!

**Comment by MK-RD** [comment]
Besides the problems already mentioned, saving after changing the name is not always guaranteed.

The used "method", which creates the configuration data (term_name) is overloaded and also therefore does not consider a reasonable data model, which considers taxonomies and term IDs.

Due to the current "implementation" there are also some dependencies to consider.

To put it in medical terms: The cancer has already spread!
